code/game/machinery/atmos_field_gen.dm

```diff
@@ -0,0 +1,289 @@
+#define ACTIVE_WANTPOWER 1
+#define ACTIVE 2
+/obj/machinery/atmos_shield_gen
+	name = "Atmospheric Shield Generator"
+	desc = "Produces an atmos shield in a line between itself and another generator with both facing the other, while active. Powered by APC. Field must not be obstructed by wall, or an atmos shield field. Will turn on after gaining power if turned off due to power loss."
+	icon = 'icons/obj/machines/atmosshieldgen.dmi'
+	base_icon_state = "atmosshield"
+	icon_state = "atmosshield"
+	density = FALSE
+	layer = ABOVE_MOB_LAYER
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION
+	circuit = /obj/item/circuitboard/machine/atmos_shield_gen
+	req_one_access = list(ACCESS_EXTERNAL_AIRLOCKS, ACCESS_ENGINE_EQUIP)
+	can_atmos_pass = ATMOS_PASS_PROC
+	power_channel = AREA_USAGE_ENVIRON
+	armor_type = /datum/armor/atmos_shield_gen
+
+	/// are we locked
+	var/locked = FALSE
+	/// are we on
+	var/on = FALSE
+	/// Max tiles between this generator and another generator. 0 would mean the generator can only do its thing if both are on the same tile.
+	var/max_range = 2
+	/// our shields
+	var/list/fields = list()
+	/// the shield generator we belong to that is actually making shields
+	var/obj/machinery/atmos_shield_gen/master = null
```

Comment:
if master is deleted is this ref cleared?

Reply:
yeah